airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py

```diff
@@ -694,8 +694,9 @@ def _adopt_completed_pods(self, kube_client: client.CoreV1Api) -> None:
                 )
             except ApiException as e:
                 self.log.info("Failed to adopt pod %s. Reason: %s", pod.metadata.name, e)
-            ti_id = annotations_to_key(pod.metadata.annotations)
-            self.running.add(ti_id)
+            else:
```

Comment:
Nit: Is there a way maybe to distinguish temporary from permanent exceptions? I do not know the API well - but this one calls for (if transient -> retry) case.

Reply:
Even if it is transient, there isn't much harm in just waiting until the next cycle to try adopting it again.

There isn't a robust way to know if its transient, afaik. You can approach it like [this](https://github.com/apache/airflow/blob/41508f23ad87bdf410978ff96e62d8057d17d9d0/airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py#L411), but I don't think it's worth it in this case.

Reply:
Oh, and I switched to `continue` instead of `else`. Same behavior, easier to read imo.